sink/datadog.go

```diff
@@ -62,9 +75,11 @@ func NewDatadog(monitorId string, opts, tags map[string]string, httpClient *http
 	}
 
 	d := &Datadog{
-		monitorId: monitorId,
-		tags:      tagList,
-		resources: resources,
+		monitorId:     monitorId,
+		tags:          tagList,
+		resources:     resources,
+		partitionSize: math.MaxInt32,
```

Comment:
Let's clarify "partition size" for future engineers who'll have to understand this code. The first confusing thing is partition (size of) _what_? We've got `MAX_PAYLOAD_SIZE` which is clear (max payload sizes are a common thing). So what's not clear is that `partitionSize` is actually something like "metrics per Send request", if I'm not mistaken? Then that plus starting with MaxInt32 makes it seems like we'll have to partition _a lot_ (when needed) because 2^32= ~4 _billion_, yet we only deal with metrics in the 10s or 1000s.

Reply:
I can update the name. My thought with starting with MaxInt32 was that we begin with _no_ partitioning of the metrics. We only start to do that if we happen to run into the case where we get a error response back from Datadog indicating the payload is too large. With compression on that is much less likely to happen in general. MaxInt32 is just a nice default to set this to instead of just arbitrarily picking a value I thought was big enough.